### ISSUE TITLE
Desktop version?

### ISSUE THREAD

**Issue by baimafeima** [monicahq/monica#531]
I have only heard about your software today. Is there any chance that you could make this available as a desktop version? The other installation options are too complicated for me. If there were a desktop version, I'd make a suggestion at a Linux-based distribution of my choice for it to be added to the repositories so I can install it.

**Comment by aejnsn** [comment]
This is a web app so it would require a ground-up rewrite to become a desktop application. @djaiss has final say on this, but I cannot fathom him constructing a desktop application.

There's good news though--you don't have to install Monica to use it. You can use the SaaS version online at monicahq.com. Just click the button in the top-right and sign up for an account. 🍰

**Comment by baimafeima** [comment]
Thanks a lot for your quick reply. I was actually looking for an application like this for a long time but I really prefer a desktop application. I don't have any need to sync my data but I value privacy a lot. Any self-hosted solution would require too much effort. I am sure there would be a great demand for having an actual desktop version.

**Comment by aejnsn** [comment]
The privacy concern you mention is one I value as well. I want to understand what would be desirable for your situation. If this were packaged in such a way to make installation easier (less involved) on a hosting provider, e.g., something like a one-click application on DigitalOcean, would that be more desirable? 

Better yet, if you're familiar with VMs on your local machine, thus wanting to run this in isolation only for yourself, would someone constructing a VM image be desirable? All you would have to do is start up a small VirtualBox image and...Boom...Monica!

**Comment by djaiss** [comment]
There is the option of using the Docker image - check the readme for instructions. This is as close as you'd have an application that you would fully control which wouldn't access the outside world.

A desktop application is really not in the short and mid term at this point.

**Comment by aejnsn** [comment]
@djaiss I don't know why I forgot the Docker image.

More details on that here: https://github.com/monicahq/monica/blob/master/docs/installation/docker.md

**Comment by baimafeima** [comment]
> If this were packaged in such a way to make installation easier (less involved) on a hosting provider, e.g., something like a one-click application on DigitalOcean, would that be more approachable?

@aejnsn Providing a one-click application for hosting providers would certainly be a good idea and I think this could increase adoption yet I would still have the privacy issue of having my data stored on someone else's computer, in that case, the hosting provider. A convincing reason would be to have the entire database end-to-end encrypted. Is this already possible?

**Comment by djaiss** [comment]
@baimafeima you are seeing it wrong.
If there were a desktop version, the simpler first step would be to point to a local database. Yes, that would mean having to run a mySQL (or else) server somewhere. Far from being ideal but at least it would be local.
Second step would be make a version that use some sort of local storage, probably SQLite or something else. However how would we deal here with all the migrations that each new version of Monica requires?
End-to-end encryptions is another beast and would make, if I understand the concept well enough, some queries really hard to do. How can you sort something when the data is encrypted? As a matter of fact, the very initial version of Monica was using an encrypted database, but it was so limiting in terms of user experience with those impossible queries I couldn't do, that I had to remove all encryption to move forward, knowing perfectly well that it would mean a data less secure.
To sum up, a desktop version is something that I want to provide at some point. First of all, I'm working on both some awesome new features for the website, and on a mobile application as well.

**Comment by djaiss** [comment]
We'd probably need to use Electron for this.

**Comment by turtles2** [comment]
We could use [this](https://github.com/cztomczak/phpdesktop) it works with laravel and we could package the whole app has one executable the only issue I had when playing around with it was getting the DB to work as it prefers SQLlite.

**Comment by trevormeier** [comment]
I also second the need for an easy-to-install desktop version. I'd be happy to use the hosted version if it had e2e encryption, otherwise I feel I'm at risk of exposing my contact's data. So until that's available, a quick and easy way to run Monica as a desktop app (Mac, in my case) would be great.

I know Monica is getting a bunch of fresh attention thanks to Derek Sivers' post (https://sivers.org/dbt) and it would be great to see it as an easy-to-use open-source competitor to the other options he recommends.

**Comment by baimafeima** [comment]
Any updates on the desktop version project?

**Comment by djaiss** [comment]
No updates at all.
The work to create a desktop app is gigantic, to say the least, and considering we are like two people working on the project (@asbiin and I), I don't see how we can find the create the desktop app ourselves and continue all the features we'd like to do on the project itself.

This is why we have an API where we can do everything on the data by the way. If someone wants to jump in and try to wrap Monica with Electron, that'd be awesome. In the meantime... 

Let's think about this a bit and see how we could make this happen.

**Comment by millette** [comment]
If you go the docker route, be aware of this *issue*: https://github.com/monicahq/monica/issues/6625#issuecomment-1992745424
